tests/test_resource.py

```diff
@@ -0,0 +1,67 @@
+#  Copyright 2019 XLAB d.o.o.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import pytest
+from redfish_client.exceptions import BlacklistedValueException, TimedOutException
+
+
+def test_get_value_of_a_key(system_resource):
+    assert system_resource.ProcessorSummary.Status.State == 'Enabled'
+
+
+def test_get_invalid_key(system_resource):
+    with pytest.raises(KeyError):
+        value = system_resource.ProcessorSummary.Status.Invalid
+
+
+def test_dig(system_resource):
+    assert system_resource.dig('ProcessorSummary', 'Status', 'State') == 'Enabled'
+
+
+def test_execute_action(connector_post_mocker, system_resource):
+    assert system_resource.execute_action('#ComputerSystem.Reset', {'ResetType': 'On'}).ok is True
+
+
+def test_execute_action_multilevel(connector_post_mocker, system_resource):
+    assert system_resource.execute_action('#ComputerSystem.CustomizedReset', {'ResetType': 'On'}).ok is True
+
+
+def test_execute_invalid_action(connector_post_mocker, system_resource):
+    with pytest.raises(KeyError):
+        system_resource.execute_action('#ComputerSystem.InvalidAction', {'ResetType': 'On'})
+
+
+@pytest.mark.parametrize(
+    'connector_get_system_mocker_with_side_effect', [(['PowerState'], ['On', 'On', 'Off'])],
+    indirect=True)
+def test_wait_for_value(connector_get_system_mocker_with_side_effect, system_resource):
+    assert system_resource.wait_for(['PowerState'], 'Off') is True
```

Comment:
Did you disable sleep somewhere or are we actually polling with sleep? You can do it simply by using decorator

```python
@mock.patch('time.sleep')
class TestWaitFor:
  def test_success(mock_sleep, ...)
  def test_fails(mock_sleep, ...)
```

Note that `@mock.patch` will mock the `time.sleep()` function so that it'll do nothing. Beware the decorator inserts a parameter as first argument to your test function, in the example above I picked name `mock_sleep` for it. You don't need to do anything with it in test function

Reply:
oh yes, of course.. And I was wondering why some tests took so long